tests/integration_tests.rs

```diff
@@ -622,6 +622,32 @@ fn complex_parallel_for_appender_loop() {
     unsafe { weld_value_free(ret_value) };
 }
 
+fn simple_for_vectorizable_loop() {
```

Comment:
Do you actually check that this vectorized the loop?

This is another reason it may be good to implement parsing -- so we can have tests that directly execute some vectorized code.